spec/integration/runners/queue/rspec_runner_spec.rb

```diff
@@ -2427,6 +2427,104 @@ def when_first_matching_example_defined(type:)
     end
   end
 
+  context 'when the RSpec split by examples is enabled AND test files are split by test examples AND slow test files are not detected (for example, the user could have passed test examples like a_spec.rb[1:1] directly using KNAPSACK_PRO_TEST_FILE_LIST_SOURCE_FILE or KNAPSACK_PRO_TEST_FILE_LIST)' do
+    before do
+      ENV['KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES'] = 'true'
+      ENV['KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN'] = ""
+      ENV['KNAPSACK_PRO_CI_NODE_TOTAL'] = '2'
+    end
+    after do
+      ENV.delete('KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES')
+      ENV.delete('KNAPSACK_PRO_SLOW_TEST_FILE_PATTERN')
+      ENV.delete('KNAPSACK_PRO_CI_NODE_TOTAL')
+    end
+
+    it 'detects test execution times correctly for individual test examples even though they are not considered as slow test files' do
+      ENV['TEST__LOG_EXECUTION_TIMES'] = 'true'
+      rspec_options = '--format d'
+
+      spec_a = Spec.new('a_spec.rb', <<~SPEC)
+        describe 'A_describe' do
+          it 'A1 test example' do
+            expect(1).to eq 1
+          end
+          it 'A2 test example' do
+            expect(1).to eq 1
+          end
+        end
+      SPEC
+
+      spec_b = Spec.new('b_spec.rb', <<~SPEC)
+        describe 'B_describe' do
+          it 'B1 test example' do
+            expect(1).to eq 1
+          end
+          it 'B2 test example' do
+            expect(1).to eq 1
+          end
+        end
+      SPEC
+
+      spec_c = Spec.new('c_spec.rb', <<~SPEC)
+        describe 'C_describe' do
+          it 'C1 test example' do
+            expect(1).to eq 1
+          end
+          it 'C2 test example' do
+            expect(1).to eq 1
+          end
+        end
+      SPEC
+
+      generate_specs(spec_helper_with_knapsack, rspec_options, [
+        [spec_a, spec_b, spec_c]
+      ])
+      stub_test_cases_for_slow_test_files([
+        "#{spec_a.path}[1:1]",
+        "#{spec_a.path}[1:2]",
+      ])
+      stub_spec_batches([
+        ["#{spec_a.path}[1:1]", spec_b.path],
+        ["#{spec_a.path}[1:2]", spec_c.path],
+      ])
+
+      actual = subject
+
+      expect(actual.stdout).to include('DEBUG -- : [knapsack_pro] Detected 0 slow test files: []')
+
+      expect(actual.stdout).to include(
+        <<~OUTPUT
+        A_describe
+          A1 test example
+
+        B_describe
+          B1 test example
+          B2 test example
+        OUTPUT
+      )
+
+      expect(actual.stdout).to include(
+        <<~OUTPUT
+        A_describe
+          A2 test example
+
+        C_describe
+          C1 test example
+          C2 test example
+        OUTPUT
+      )
+
+      expect(actual.stdout.scan(/A1 test example/).size).to eq 1
+      expect(actual.stdout.scan(/A2 test example/).size).to eq 1
+
+      expect(actual.stdout).to include('6 examples, 0 failures')
+
+      expect(actual.stdout).to include('[INTEGRATION TEST] test_files: 4, test files have execution time: true')
```

Comment:
If the bug was not solved this assertion would fail because in the logs it would be `'[INTEGRATION TEST] test_files: 4, test files have execution time: false'`?

Reply:
Yes.

It would be:

```
'[INTEGRATION TEST] test_files: 5, test files have execution time: false'
```

5 test files would be measured in the old approach:

```
a_spec.rb with time
a_spec.rb[1:1] with 0s
a_spec.rb[1:2] with 0s
b_spec.rb with time
c_spec.rb with time
```